src/machine/machined-varlink.c

```diff
@@ -510,6 +512,37 @@ static int vl_method_list(sd_varlink *link, sd_json_variant *parameters, sd_varl
         return sd_varlink_error(link, "io.systemd.Machine.NoSuchMachine", NULL);
 }
 
+static int lookup_machine_and_call_method(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata, sd_varlink_method_t method) {
+        static const sd_json_dispatch_field dispatch_table[] = {
+                { "name", SD_JSON_VARIANT_STRING, sd_json_dispatch_const_string, 0, SD_JSON_MANDATORY },
+                VARLINK_DISPATCH_POLKIT_FIELD,
```

Comment:
Similar to list, maybe better to take `pid` ?

Reply:
Do you suggest to expand functionality so it becomes possible to Unregister/Terminate/Kill not only by name, but by pid as well?

Reply:
Yeah we should support that for all methods that look up a machine